<launch>
	<param name="/use_sim_time" value="true"/>


    <arg name="block_1" default="" />  <!-- Set to nothing or "active" -->
    <arg name="block_2" default="" />
    <arg name="movement_noise" default="0" />


    <node pkg="map_server" type="map_server" name="map_server" args=" $(find explorer)/files/whippu_dungeon.map.yaml" output="screen"></node>
    <node pkg="amcl" type="amcl" name="localization">
        <!--<param name="use_map_topic" value="true" />-->
        <!--<remap from="map" to="map"/>-->
    </node>
    <include file="$(find explorer)/config/move_base_copy.launch.xml"/>

    <include file="$(find explorer)/launch/turtlebot_world_copy.launch">
	    <arg name="world_file" value="$(env HOME)/gazebo/whippu_dungeon.world" />
    </include>

    <node name="object_manager" pkg="explorer" type="object_manager.py">
        <param name="block_1" value="$(arg block_1)" />
        <param name="block_2" value="$(arg block_2)" />
    </node>

    <node name="temporal_data_gatherer" pkg="explorer" type="world_publisher.py" />
    <node name="temporal_map_publisher" pkg="explorer" type="temporal_map_publisher.py" />

    <include file="$(find explorer)/launch/cartographer_isolated.launch" />
</launch>

<!-- rosbag record /scan /odom /tf_static /tf /mobile_base/sensors/imu_data -O temporal.bag -->